pkg/controller/cluster/cluster_reconcile_test.go

```diff
@@ -452,46 +455,42 @@ func TestReconcileCluster_getServiceAccountToken(t *testing.T) {
 }
 
 func TestReconcileCluster_unmanagedGitRepo(t *testing.T) {
```

Comment:
As an alternative (or in addition?) we could also name this with the given/when/then syntax. It makes it more clear what the test actually does
```suggestion
func TestReconcileCluster_GivenClusterWithUnmanagedGitRepo_WhenReconcileCluster_Then???(t *testing.T) { // what exactly is expected here?
```

Reply:
If we realy want to do BDD, then we would need to switch to another testing framework. Doing this with gotest, feels awekward. Let us discuss this seperateley and leave this for now.